tests/evm/test_call_staticcall.py

```diff
@@ -181,31 +213,42 @@ def test_call(
         )
     )
 
+    is_static, call_id, rw_counter, next_program_counter, stack_pointer = (
+        (False, 24, 24, 232, 1017) if is_call else (True, 23, 23, 199, 1018)
+    )
+
     # fmt: off
     rw_dictionary = (
-        RWDictionary(24)
+        RWDictionary(rw_counter)
         .call_context_read(1, CallContextFieldTag.TxId, 1)
         .call_context_read(1, CallContextFieldTag.RwCounterEndOfReversion, caller_ctx.rw_counter_end_of_reversion)
         .call_context_read(1, CallContextFieldTag.IsPersistent, caller_ctx.is_persistent)
         .call_context_read(1, CallContextFieldTag.CalleeAddress, caller.address)
-        .call_context_read(1, CallContextFieldTag.IsStatic, False)
+        .call_context_read(1, CallContextFieldTag.IsStatic, is_static)
```

Comment:
This `is_static` is actually the context of current call, so it's not always equal to `opcode == STATICCALL`. Would you like to add an extra test matrix `is_static = [True, False]` in `gen_testing_data`?

Reply:
It seems that `is_static` of current call is useless and could not be constrained.

I deleted this lookup (seems unnecessary to add `is_statics = [True, False]` in `gen_test_data`?), and add a lookup for `is_static` from `callee` and constrain it as:
```
# Get callee is_static and Verify is_static == 0 for opcode CALL, and is_static == 1 for opcode STATICCALL.
is_static = instruction.call_context_lookup(
  CallContextFieldTag.IsStatic, call_id=callee_call_id
)
instruction.constrain_equal(is_call + is_static, FQ(1))
```
And updated test code.

@han0110 Could you help review if it is correct when you have time? Thanks a lot.

Reply:
@han0110 if current call is static call, still need to constrain its subcall is also static?

Reply:
> if current call is static call, still need to constrain its subcall is also static?

yes, so we should initialize `is_static = current_call_context.is_static || opcode = STATICCALL`, and use it to check with `has_value` and constrain the next call context to have `is_static`.

Reply:
Fixed to initialize and constrain `is_static` as:
```
is_static = instruction.select(
    is_call, instruction.call_context_lookup(CallContextFieldTag.IsStatic), FQ(1)
)

...

# Verify is_static == 1 for opcode STATICCALL.
instruction.constrain_zero((1 - is_call) * (1 - is_static.expr()))
```

In test cases, set `is_static` as:
```
is_call = opcode == Opcode.CALL
is_static = not is_call or stack.value == 0
```